aspnet/fundamentals/startup.rst

```diff
@@ -1,8 +1,196 @@
+
+The Configure method
+^^^^^^^^^^^^^^^^^^^^
+
+The ``Configure`` method is used to specify how the ASP.NET application will respond to individual HTTP requests. At its simplest, you can configure every request to receive the same response. However, most real-world applications require more functionality than this. More complex sets of pipeline configuration can be encapsulated in :doc:`middleware` and added using extension methods on `IApplicationBuilder <https://github.com/aspnet/HttpAbstractions/blob/8703e2d7f21d09b50d20cc764e11d6bb0268aad2/src/Microsoft.AspNet.Http.Abstractions/IApplicationBuilder.cs>`_. This helps keep the ``Configure`` method concise and easy to follow, and allows for good `Separation of Concerns <http://deviq.com/separation-of-concerns/>`_. 
+
+Your ``Configure`` method must accept an ``IApplicationBuilder`` parameter. Additional parameters, like ``IHostingEnvironment`` and ``ILoggerFactory`` may also be specified, in which case these services will be injected by the server if they are available. In the following example from the default web site template, you can see several extension methods are used to configure the pipeline with support for BrowserLink, error pages, static files, ASP.NET MVC, and Identity.
+
+.. literalinclude:: ../../common/samples/WebApplication1/src/WebApplication1/Startup.cs
+	:language: c#
+	:linenos:
+	:lines: 88-134
+	:dedent: 8
+	:emphasize-lines: 12-14,20,24,27,37
+
+You can see what each of these extensions does by examining the source. For instance, the ``UseMvc`` extension method is defined in `BuilderExtensions <https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNet.Mvc/BuilderExtensions.cs>`_, available on GitHub. Its primary responsibility is to ensure MVC was added as a service (in ``ConfigureServices``) and to correctly set up routing for an ASP.NET MVC application. We'll examine the specifics of how to configure your application's request pipeline later in this article.
+
+.. note:: To see an example of how ``IApplicationBuilder`` is implemented, have a look at `Microsoft.AspNet.Http.ApplicationBuilder.cs <https://github.com/aspnet/HttpAbstractions/blob/eb0fe6a92ada31440e6db44d23b64f4d3d7240c0/src/Microsoft.AspNet.Http/ApplicationBuilder.cs>`_.
+
+
+The ConfigureServices method
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Your `Startup` class can optionally include a ``ConfigureServices`` method, which is called before ``Configure``. This is important, because some features like ASP.NET MVC require certain services to be added in ``ConfigureServices`` before they can be wired up to the request pipeline.
+
+Just as with ``Configure``, it is recommended that features that require substantial setup within ``ConfigureServices`` be wrapped up in extension methods on ``IApplicationBuilder``. You can see in this example from the default web site template that several ``Add[Something]`` extension methods are used to configure the app to use Entity Framework, Identity, and MVC:
```

Comment:
I think this should be

> be wrapped up in extension methods on ``IServiceCollection``

rather than 

> be wrapped up in extension methods on ``IApplicationBuilder``